students/ssankura/lesson03/mailroom.py

```diff
@@ -0,0 +1,146 @@
+	- If the user enters "list" , display all the donors 
+'''
+def sendThankYou():
+	loop = True
+	while (loop):		
+		response = input("Enter the Full Name of the Donor > ")
+		if (response != "") and (response != "list"):
+			if not(response in donorsList):
+				#Add the new donor to the Donors List
+				donorsList.append(response)
+				#do something and break from the While loop
+			donor = response
+			response = input("Enter the donation amount > ")
+			amount = float(response)
+			donorsAmountList.append((donor, amount))
+			print (f"Hello {donor}, \n \t Thank You for the generous donation of ${amount}. We appreciate your effort to help out with the cause that we have been supporting. \n From, \n ABC Charity")
+			loop = False
+		else:
+			print ("List of Donors:")
+			print (("{:30} "*len(donorsList)).format(*donorsList))
+
+
+'''
+Create a report in the format specified in the requirements - as shown below
+Donor Name                | Total Given | Num Gifts | Average Gift
+------------------------------------------------------------------
+William Gates, III         $  653784.49           2  $   326892.24
+Mark Zuckerberg            $   16396.10           3  $     5465.37
+Jeff Bezos                 $     877.33           1  $      877.33
+Paul Allen                 $     708.42           3  $      236.14
+
+Compute the Donor info along with total donation amount, number of gifts and average amount
+	- store the info as tuples in a list - donorInfoList
+Print the Data in the specified format
+'''
+def createReport():
+	donorInfoList = []
+	for donor in donorsList:
+		total = 0 
+		numGifts = 0
+		for item in donorsAmountList:
+			donorNameInItem = item[0]
+			if donorNameInItem == donor:
+				total  = total + int(item[1])
+				numGifts = numGifts + 1
+		averageAmount = total/numGifts
+		donorInfoList.append ((donor,total,numGifts,averageAmount))
+	sp1 = " " * 19
+	strHeader = "Donor Name " + sp1 + " | Total Given  | Num Gifts | Average Gift"
+	print (strHeader)
+	strDash = "-" * len(strHeader)
+	print (strDash)
+	for tmpitem in donorInfoList:
+		print(f"{tmpitem[0]:30}  ${tmpitem[1]:10.2f}  {tmpitem[2]:10}    ${tmpitem[3]:10.2f}")
+
+
+'''
+Populate data in the global variables - donorsList and donorsAmountList
+'''
+def populateData():
+	donorsList.append("Mark Zuckerberg")
+	donorsList.append("Jeff Bezos")
+	donorsList.append("Paul Allen")
+	donorsList.append("Joanne K Rowling")
+	donorsAmountList.append(("Mark Zuckerberg",5000.50))
+	donorsAmountList.append(("Jeff Bezos",700.90))
+	donorsAmountList.append(("Mark Zuckerberg",1089.90))
+	donorsAmountList.append(("Joanne K Rowling",9878.12))
+	donorsAmountList.append(("Jeff Bezos",900.99))
+	donorsAmountList.append(("Mark Zuckerberg",750.45))
+	donorsAmountList.append(("Joanne K Rowling",910.75))
+	donorsAmountList.append(("Paul Allen",11100.80))
+	donorsAmountList.append(("Joanne K Rowling",1100.80))
```

Comment:
Would it be easier to just define it all in one shot:
`donar_list = ["Mark", "Jeff", "Paul", "Joanne"]`
The same with `donar_amount_list`.
but honestly you should probably be using dictionary
```
{
     "Mark": [5000.50, 1089.90, 750.45],
     "Jeff": [700.90, 900.00],
}```
easier to visualize and iterate through.